iree/hal/driver_registry.h

```diff
@@ -102,6 +102,17 @@ typedef struct iree_hal_driver_registry_s iree_hal_driver_registry_t;
 IREE_API_EXPORT iree_hal_driver_registry_t* IREE_API_CALL
 iree_hal_driver_registry_default();
 
+// Creates and initializes an empty device driver registry
+//
+// Thread-safe.
+IREE_API_EXPORT iree_status_t IREE_API_CALL iree_hal_driver_registry_create(
```

Comment:
`_create` is reserved for reference counted objects (we roughly follow the core foundation naming and rules, so the names are meaningful: https://developer.apple.com/library/archive/documentation/CoreFoundation/Conceptual/CFMemoryMgmt/Concepts/Ownership.html)

Since this is just a non-refcounted instance this should be `_allocate` and the free should just be `_free` (no need for `unregister_and_`).